tags.go

```diff
@@ -16,10 +16,32 @@ func (t tagSet) Len() int           { return len(t) }
 func (t tagSet) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
 func (t tagSet) Less(i, j int) bool { return t[i].dimension < t[j].dimension }
 
+func fixTags(tags map[string]string) map[string]string {
+	noEmpty := true
+	for _, v := range tags {
+		if v == "" {
+			noEmpty = false
+			break
+		}
+	}
+	if noEmpty {
+		return tags
+	}
+	newTags := make(map[string]string)
```

Comment:
nit, `make(map[string]string, len(tags) - 1)` since we know the size and will save allocs if `len(tags)` is larger than the default map size of 8.